libmamba/src/api/pip_utils.cpp

```diff
@@ -0,0 +1,174 @@
+#include <fmt/color.h>
+#include <fmt/format.h>
+#include <fmt/ostream.h>
+#include <fmt/ranges.h>
+#include <reproc++/run.hpp>
+#include <reproc/reproc.h>
+
+#include "mamba/api/install.hpp"
+#include "mamba/core/activation.hpp"
+#include "mamba/core/context.hpp"
+#include "mamba/core/util.hpp"
+#include "mamba/fs/filesystem.hpp"
+#include "mamba/util/environment.hpp"
+
+#include "tl/expected.hpp"
+
+#include "pip_utils.hpp"
+
+namespace mamba
+{
+    namespace
+    {
+        tl::expected<command_args, std::runtime_error> get_pip_install_command(
+            const std::string& name,
+            const std::string& target_prefix,
+            const fs::u8path& spec_file,
+            bool update
+        )
+        {
+            const auto python = util::which_in("python", get_path_dirs(target_prefix)).string();
```

Comment:
Btw, I think we should restore the previous lambda function here, as it should take into consideration the changes in the `target_prefix` and compute the python path on demand.